deps/chakrashim/src/v8string.cc

```diff
@@ -91,8 +91,9 @@ int String::Write(uint16_t *buffer, int start, int length, int options) const {
   size_t count = 0;
   if (JsCopyStringUtf16((JsValueRef)this, start, length,
                          buffer, &count) == JsNoError) {
-    if (!(options & String::NO_NULL_TERMINATION)) {
-      buffer[count] = 0;
+    if (!(options & String::NO_NULL_TERMINATION) &&
+        (length == -1 || count < length)) {
```

Comment:
This is a bug in xplat as well, but luckily nobody seems to hit it.  This matches the behavior of the V8 code, but potentially leaves a buffer unterminated if there's not enough room in the buffer to fit the string and the null-terminator.  My preference would likely be to force a null in at the end unless null termination is off, but that would require changing more code and potentially break in the future.